packages/babel-core/src/transformation/file/options/config.js

```diff
@@ -33,6 +33,12 @@ module.exports = {
     description: "retain line numbers - will result in really ugly code"
   },
 
+  retainParens: {
```

Comment:
We don't have to add an entry here since this is for top level babel options (assuming we are ok with just using `{generatorOpts: { retainParens: true } }`